lms/djangoapps/edraak_certificates/views.py

```diff
@@ -1,59 +1,93 @@
 
-
 logger = logging.getLogger(__name__)
 
-@login_required
-def issue(request, course_id):
-    return render_to_response('edraak_certificates/issue.html', {
-        'course_id': course_id,
-    })
-
 
 @transaction.non_atomic_requests
 @login_required
-def view(request, course_id):
-    user = request.user
+def issue(request, course_id):
+    student = request.user
 
+    # Extracts CourseLocator Object from course ID.
     course_key = locator.CourseLocator.from_string(course_id)
+    # Extract the CourseOverview object of the course.
     course = modulestore().get_course(course_key)
 
-    if is_certificate_allowed(user, course) and is_student_pass(user, request, course_id):
-        template = 'edraak_certificates/view.html'
-    else:
-        template = 'edraak_certificates/fail.html'
+    # Check the status of GeneratedCertificate in XQueue.
+    certificate_status = \
+        certificate_status_for_student(student, course.id)['status']
+
+    template = None
+
+    # If user didn't issue or pass certificate, or an error happened
+    # in the generation.
+    if certificate_status in [CertificateStatuses.unavailable,
+                              CertificateStatuses.notpassing,
+                              CertificateStatuses.error]:
+        # If the course is certifiable and the user has passed it.
+        if is_certificate_allowed(student, course) and \
+                is_student_pass(student, request, course_id):
+            # generate the certificate
+            generate_user_certificates(student, course.id,
+                                       course=course)
+            template = 'edraak_certificates/issue.html'
+        else:
+            # Didn't pass the course template.
+            template = 'edraak_certificates/fail.html'
+
+    elif certificate_status == CertificateStatuses.downloadable:
+        cert = GeneratedCertificate.objects.get(
+            user_id=student.id,
+            course_id=course_key,
+            status=CertificateStatuses.downloadable
+        )
+        return render_cert_by_uuid(request, cert.verify_uuid)
 
     course_end_date = course.end or date.today()
-    course_expiration_date = course_end_date + relativedelta(years=settings.BAYT_VALID_YEARS)
+    course_expiration_date = course_end_date + relativedelta(
+        years=settings.BAYT_VALID_YEARS)
 
     return render_to_response(template, {
-        'user': user,
-        'cert_course': course,  # Course name is set to `cert_course` to avoid header design bug
+        'course_id': course_id,
+        'user': student,
+        'cert_course': course,
         'cert_end': course_expiration_date.strftime('%m-%Y')
     })
 
 
+@transaction.non_atomic_requests
+@login_required
+def check_status(request, course_id):
+    course_key = locator.CourseLocator.from_string(course_id)
+    status = certificate_downloadable_status(request.user, course_key)
+    return JsonResponse(data=status)
```

Comment:
Let's wrap the view with [@api_view()](http://www.django-rest-framework.org/api-guide/views/#api_view) instead.